buffer/src/main/java/io/netty5/buffer/internal/CleanerPool.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2022 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty5.buffer.internal;
+
+import io.netty5.util.concurrent.FastThreadLocal;
+import io.netty5.util.internal.SystemPropertyUtil;
+import io.netty5.util.internal.ThreadExecutorMap;
+import io.netty5.util.internal.logging.InternalLogger;
+import io.netty5.util.internal.logging.InternalLoggerFactory;
+
+import java.lang.ref.Cleaner;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.IntStream;
+
+/**
+ * Cleaner allocation strategy used to assign cleaners to event-loop and external threads.
```

Comment:
```suggestion
 * Cleaner allocation strategy used to assign cleaners to event-loops and external threads.
```

Reply:
done, also fixed another type (`shared cleaner pool` instead of `shared thread pool`).